spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -0,0 +1,649 @@
+        partition.format,
+        spec,
+        conf.get(),
+        metricsConfig);
+  }
+
+  /**
+   * Returns the data files in a partition by listing the partition location.
+   *
+   * For Parquet and ORC partitions, this will read metrics from the file footer. For Avro partitions,
+   * metrics are set to null.
+   *
+   * @param partition     partition key, e.g., "a=1/b=2"
+   * @param uri           partition location URI
+   * @param format        partition format, avro or parquet
+   * @param conf          a Hadoop conf
+   * @param metricsConfig a metrics conf
+   * @return a List of [[DataFile]]
+   */
+  public static List<DataFile> listPartition(
+      Map<String, String> partition,
+      String uri,
+      String format,
+      PartitionSpec spec,
+      Configuration conf,
+      MetricsConfig metricsConfig) throws IOException {
+
+    if (format.contains("avro")) {
+      return listAvroPartition(partition, uri, spec, conf);
+    } else if (format.contains("parquet")) {
+      return listParquetPartition(partition, uri, spec, conf, metricsConfig);
+    } else if (format.contains("orc")) {
+      // TODO: use MetricsConfig in listOrcPartition
+      return listOrcPartition(partition, uri, spec, conf);
+    } else {
+      throw new UnsupportedOperationException("Unknown partition format: " + format);
+    }
+  }
+
+  private static List<DataFile> listAvroPartition(
+      Map<String, String> partitionPath,
+      String partitionUri,
+      PartitionSpec spec,
+      Configuration conf) throws IOException {
+
+    Path partition = new Path(partitionUri);
+    FileSystem fs = partition.getFileSystem(conf);
+    return Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
+        .filter(FileStatus::isFile)
+        .map(stat -> {
+          Metrics metrics = new Metrics(-1L, null, null, null);
+          String partitionKey = spec.fields().stream()
+              .map(PartitionField::name)
+              .map(name -> String.format("%s=%s", name, partitionPath.get(name)))
+              .collect(Collectors.joining("/"));
+
+          return DataFiles.builder(spec)
+              .withPath(stat.getPath().toString())
+              .withFormat("avro")
+              .withFileSizeInBytes(stat.getLen())
+              .withMetrics(metrics)
+              .withPartitionPath(partitionKey)
+              .build();
+
+        }).collect(Collectors.toList());
+  }
+
+  private static List<DataFile> listParquetPartition(
+      Map<String, String> partitionPath,
+      String partitionUri,
+      PartitionSpec spec,
+      Configuration conf,
+      MetricsConfig metricsSpec) throws IOException {
+
+    Path partition = new Path(partitionUri);
+    FileSystem fs = partition.getFileSystem(conf);
+
+    return Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
+        .filter(FileStatus::isFile)
+        .map(stat -> {
+          final Metrics metrics;
+          try {
+            //noinspection deprecation
```

Comment:
Does this comment still apply in Java? Isn't it specific to Scalastyle?

Reply:
This was an intellij suppression for the deprecated usage of `ParquetFileReader.readFooter`. I removed it now.